core/store/src/trie/mod.rs

```diff
@@ -757,6 +760,93 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    /// Given a root and a key, it returns a MerkleProof for that key.
+    ///
+    ///  Whenever a proof is not found, an empty vector is returned.
```

Comment:
what does it mean that 'proof is not found'? from what I see, if the key doesn't exist in a TRIE, you'll return StorageError, right? 

(current code would return empty/None only of root is an empty tree -- but I'd argue that in this case, we should also return a StorageError - as this is the same case as key not being present).